staging/src/k8s.io/client-go/util/consistencydetector/data_consistency_detector.go

```diff
@@ -75,8 +74,8 @@ func CheckDataConsistency[T runtime.Object, U any](ctx context.Context, identity
 	sort.Sort(byUID(listItems))
 	sort.Sort(byUID(retrievedItems))
 
-	if !cmp.Equal(listItems, retrievedItems) {
-		klog.Infof("previously received data for %s is different than received by the standard list api call against etcd, diff: %v", identity, cmp.Diff(listItems, retrievedItems))
+	if !reflect.DeepEqual(listItems, retrievedItems) {
```

Comment:
```suggestion
	if !apiequality.Semantic.DeepEqual(listItems, retrievedItems) {
```

(https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/equality/semantic.go)

This is safer for comparing runtime.Object.